classes/core/APIRouter.php

```diff
@@ -107,7 +107,11 @@ public function route($request)
 
         $handler = require('./' . $sourceFile);
         $this->setHandler($handler);
-        $handler->getApp()->run();
+        
+        $handlerApp = $handler->getApp();
+        if ($handlerApp !== null) {
```

Comment:
What would cause `$handlerApp` to be null? (If it's a possible case, then shouldn't there be error handling here?)

Reply:
This become a case with applying and matching laravel routing system which applied [here](https://github.com/pkp/pkp-lib/pull/9176/files#diff-d6cd68c8ec4cd5788ad82e8f44a3fbe49186121c580151a3ac59315742e0dee6R66-R110) . Here we are returning early if match with any laravel routes or if any routes defined(as for not we have those define in each controller class) . so `app` never get to initialized . that why we need that check. It will become unnecessary once we refactor all handler to controller and remove the slim from the system .

Reply:
OK, please add a `// FIXME pkp/pkp-lib#7698` here so we don't forget before merging (and/or a checkbox in the issue description)!

Reply:
Added .